optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/list/ListChangeMove.java

```diff
@@ -0,0 +1,223 @@
+     *
+     * <h4>ListChangeMove anatomy</h4>
+     *
+     * <pre>
+     * {@code
+     *                             ┌ destinationEntity
+     *                             │   ┌ destinationIndex
+     *                             ↓   ↓
+     *                A {Ann[0]}->{Bob[2]}
+     *                ↑  ↑   ↑
+     * planning value ┘  │   └ sourceIndex
+     *                   └ sourceEntity
+     * }
+     * </pre>
+     *
+     * <h4>Example 1 - source and destination entities are different</h4>
+     *
+     * <pre>
+     * {@code
+     * GIVEN
+     * Ann.tasks = [A, B, C]
+     * Bob.tasks = [X, Y]
+     *
+     * WHEN
+     * ListChangeMove: A {Ann[0]->Bob[2]}
+     *
+     * THEN
+     * Ann.tasks = [B, C]
+     * Bob.tasks = [X, Y, A]
+     * }
+     * </pre>
+     *
+     * <h4>Example 2 - source and destination is the same entity</h4>
+     *
+     * <pre>
+     * {@code
+     * GIVEN
+     * Ann.tasks = [A, B, C]
+     *
+     * WHEN
+     * ListChangeMove: A {Ann[0]->Ann[2]}
+     *
+     * THEN
+     * Ann.tasks = [B, C, A]
+     * }
+     * </pre>
+     *
+     * @param variableDescriptor descriptor of a list variable, for example {@code Employee.taskList}
+     * @param sourceEntity planning entity instance from which a planning value will be removed, for example "Ann"
+     * @param sourceIndex index in sourceEntity's list variable from which a planning value will be removed
+     * @param destinationEntity planning entity instance to which a planning value will be moved, for example "Bob"
+     * @param destinationIndex index in destinationEntity's list variable where the moved planning value will be inserted
+     */
+    public ListChangeMove(
+            ListVariableDescriptor<Solution_> variableDescriptor,
+            Object sourceEntity, int sourceIndex,
+            Object destinationEntity, int destinationIndex) {
+        this.variableDescriptor = variableDescriptor;
+        this.sourceEntity = sourceEntity;
+        this.sourceIndex = sourceIndex;
+        this.destinationEntity = destinationEntity;
+        this.destinationIndex = destinationIndex;
+    }
+
+    public Object getSourceEntity() {
+        return sourceEntity;
+    }
+
+    public int getSourceIndex() {
+        return sourceIndex;
+    }
+
+    public Object getDestinationEntity() {
+        return destinationEntity;
+    }
+
+    public int getDestinationIndex() {
+        return destinationIndex;
+    }
+
+    public Object getMovedValue() {
+        return variableDescriptor.getElement(sourceEntity, sourceIndex);
+    }
+
+    // ************************************************************************
+    // Worker methods
+    // ************************************************************************
+
+    @Override
+    public boolean isMoveDoable(ScoreDirector<Solution_> scoreDirector) {
+        // TODO maybe remove this because no such move should be generated
```

Comment:
OTOH, moves are also created outside OptaPlanner, for example in the examples' UI (is this a good idea?).